### ISSUE TITLE
Twilight Forest biomes generating in BOP overworld

### ISSUE THREAD

**Issue by Taschenmogul** [Glitchfiend/BiomesOPlenty#334]
The title pretty much says it all - when using the Twilight Forest mod together with BiomesOPlenty I occasionally get TF biomes generating in the overworld (with BOP worldtype chosen).
I am quite puzzled as nobody else seems to have reported that issue, so at first I thought that I personally might have done something wrong. Having had a friend check on this and verify the bug I am now comfortable to report it to you.

I have tried several biome IDs, two times I remapped the complete TF ID range, then again I tested it with standard IDs. None of those IDs seemed to overlap or conflict and I am positive that there is no ID conflict that I could be aware of.
There was an issue though where TF auto-mapped its IDs so that some of them conflicted BOP biome IDs, but this only seemed to happen when other mods were present that brought biomes into the game and it didn't happen anymore with only TF and BOP installed.
As of now I have seen three TF biomes generating in the overworld, the "Enchanted Forest", "Thornlands" and "Dark Forest" biomes, but only occasionally. 

Technical details/mod versions:
I have had this issue with several versions of Forge and several versions of BOP (at least .985, .986 and .987, but I believe I also had this issue with a version from before the fix that broke "seed-compatibility") and tested it with Java 7 versions 1.7.0_55 and 1.7.0_67 and Java 8 version 1.8.0-b132 (all 64-Bit). I have been able to consistently reproduce this with the following versions, using default configs:

Java 1.7.0_67 64-Bit, MultiMC 5 0.4.2, Minecraft 1.7.10, Forge 10.13.0.1207, Twilight Forest 2.2.3, BiomesOPlenty 2.1.0.987.
Starting a world with BOP world type and the following seed

-2845210373478548811

there is a TF Enchanted Forest biome at coordinates X = -1900, Y, Z = 2450.

I am not sure if this issue is caused by Biomes O Plenty or not, I guess I will post a link to this report at the Twilight Forest forum thread or Github page too. Please still check up on this though! =D

**Comment by Ogliss** [comment]
Issue still occuring in MC 1.7.10 with BOP and the FTB Infinity 1.2.0 modpack

**Comment by crunkazcanbe** [comment]
I am getting this in chat biome id conflict detected . Fix by editing the config file . Well every time I make a pack I go in and make sure every enchant, potion ,and biome id of every mod is different , and no matter how many times I change the ids for twilight forest I get this in chat . I no for sure there are no conflicts unless its a id that can not be changed in a config file .

**Comment by Sakata-MC** [comment]
So since we know the behavior, is there any biome ID numbers we can use to resolve the issue?  This seriously hoses several major modpacks out there.

**Comment by maruohon** [comment]
Kind of, yes. What I did, is I put all the biomes that are not overworld biomes, into the lower range. Because vanilla will not use those, it won't mess up stuff by trying to use biomes from the upper range as sub biomes to those. So basically I had to make sure that any biomes that can generate in the overworld, either don't have a matching "sub" biome at ID + 128, or the sub biome is actually a meaningful one.

In case it's useful, here is my biome ID list for my own pack I did a while back: http://pastebin.com/Q3tm6kz9
Of course I can't guarantee this works as intended, but at least with this set of IDs I didn't encounter any mis-placed biomes in my own testing anymore.
